### PR TITLE
DBZ-2611 Provide separate project version for releases

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2611

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,6 +37,53 @@
         <vitess.vtctld.grpc.port>15999</vitess.vtctld.grpc.port>
     </properties>
 
+    <dependencyManagement>
```
Comment (keweishang) [inline pom.xml:40]: The `debezium-parent` module has already `dependencyManagement`, is it a convention to redo the same thing here?

File: pom.xml [hunk 2]
```diff
@@ -36,6 +37,53 @@
...
+        <dependencies>
+            <!-- Debezium artifacts -->
+            <dependency>
+                <groupId>io.debezium</groupId>
+                <artifactId>debezium-connector-vitess</artifactId>
```
Comment (keweishang) [inline pom.xml:45]: What is the reason to add `debezium-connector-vitess` dependency in `dependencyManagement`?

File: pom.xml [hunk 3]
```diff
@@ -36,6 +37,53 @@
...
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.debezium</groupId>
+                <artifactId>debezium-connector-mysql</artifactId>
```
Comment (keweishang) [inline pom.xml:50]: We don't need `debezium-connector-mysql` dependency.

File: pom.xml [hunk 4]
```diff
@@ -36,6 +37,53 @@
...
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.debezium</groupId>
+                <artifactId>debezium-core</artifactId>
+                <version>${version.debezium}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.debezium</groupId>
+                <artifactId>debezium-ddl-parser</artifactId>
```
Comment (keweishang) [inline pom.xml:60]: No need for `debezium-ddl-parser`.

**Comment by jpechane** [comment]
@keweishang Thanks for the review, I reduced the unnecessary dependencies.
The need for dependency management is due to release process. Mven release plugin requires the module to be of `-SNAPHSOT` version but if we don't add the plugin management it will mena that Debezium dependencies will be in `-SNAPSHOT` version which is also disallowed. So we change the parent version to for example `1.4.0.Final`, the module version is 1.4.0-SNAPSHOT and the Debezium dependencies are overridden in dependency management to be of the parent one - the final.

**Comment by keweishang** [comment]
@jpechane Thanks for the explanation of the reason of using dependency management. It is for being aligned with maven release plug-in's behavior, I see.

**Review by keweishang — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (gunnarmorling) [reply pom.xml:40]: @jpechane, good point, I don't think we need those repeated here?

Comment (jpechane) [reply pom.xml:40]: @gunnarmorling Explained in https://github.com/debezium/debezium-connector-vitess/pull/3#issuecomment-703583814

Comment (gunnarmorling) [reply pom.xml:40]: Ah, I see. Warrants a comment there in the _pom.xml_. I'll add it and merge. Thx!

**Pull request merged**